control-plane-pki-operator/certrotationcontroller/certrotationcontroller.go

```diff
@@ -0,0 +1,153 @@
+	hostedControlPlane *hypershiftv1beta1.HostedControlPlane,
+	kubeClient kubernetes.Interface,
+	hypershiftClient hypershiftv1beta1client.HostedControlPlanesGetter,
+	configInformer configinformers.SharedInformerFactory,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+	day time.Duration,
+) (*CertRotationController, error) {
+	ret := &CertRotationController{
+		recorder: eventRecorder,
+		cachesToSync: []cache.InformerSynced{
+			configInformer.Config().V1().Networks().Informer().HasSynced,
+			configInformer.Config().V1().Infrastructures().Informer().HasSynced,
+		},
+	}
+
+	rotationDay := defaultRotationDay // TODO(deads2k): is this logic we want to persist? looks temporary ...
+	if day != time.Duration(0) {
+		rotationDay = day
+		klog.Warningf("!!! UNSUPPORTED VALUE SET !!!")
+		klog.Warningf("Certificate rotation base set to %q", rotationDay)
+	} else {
+		// for the development cycle, make the rotation 60 times faster (every twelve hours or so).
+		// This must be reverted before we ship
+		rotationDay = rotationDay / 60
+	}
+
+	ownerRef := &metav1.OwnerReference{
+		APIVersion: hypershiftv1beta1.GroupVersion.String(),
+		Kind:       "HostedControlPlane",
+		Name:       hostedControlPlane.Name,
+		UID:        hostedControlPlane.UID,
+	}
+
+	rotatorName := "CustomerAdminKubeconfigSigner"
+	certRotator := certrotation.NewCertRotationController(
+		rotatorName,
+		certrotation.RotatedSigningCASecret{
+			Namespace:     hostedControlPlane.Namespace,
+			Name:          manifests.CustomerSystemAdminSigner(hostedControlPlane.Namespace).Name,
+			Validity:      7 * rotationDay,
+			Refresh:       2 * rotationDay,
+			Informer:      kubeInformersForNamespaces.InformersFor(hostedControlPlane.Namespace).Core().V1().Secrets(),
+			Lister:        kubeInformersForNamespaces.InformersFor(hostedControlPlane.Namespace).Core().V1().Secrets().Lister(),
+			Client:        kubeClient.CoreV1(),
+			EventRecorder: eventRecorder,
+			Owner:         ownerRef,
+		},
+		certrotation.CABundleConfigMap{
+			Namespace:     hostedControlPlane.Namespace,
+			Name:          manifests.CustomerSystemAdminSignerCA(hostedControlPlane.Namespace).Name,
+			Informer:      kubeInformersForNamespaces.InformersFor(hostedControlPlane.Namespace).Core().V1().ConfigMaps(),
+			Lister:        kubeInformersForNamespaces.InformersFor(hostedControlPlane.Namespace).Core().V1().ConfigMaps().Lister(),
+			Client:        kubeClient.CoreV1(),
+			EventRecorder: eventRecorder,
+			Owner:         ownerRef,
+		},
+		certrotation.RotatedSelfSignedCertKeySecret{
+			Namespace: hostedControlPlane.Namespace,
+			Name:      manifests.CustomerSystemAdminClientCertSecret(hostedControlPlane.Namespace).Name,
+			Validity:  36 * rotationDay / 24,
+			Refresh:   6 * rotationDay / 24,
+			CertCreator: &certrotation.ClientRotation{
+				UserInfo: &user.DefaultInfo{Groups: []string{"system:masters"}}, // TODO: name, uid - what to set?
```

Comment:
@deads2k

Reply:
username of "recognizeable prefix" + crypto-rand-number/uid.  Fully crypto-rand UID.